include/s3/hash.h

```diff
@@ -0,0 +1,45 @@
+/*-------------------------------------------------------------------------
+ *
+ * hash.h
```

Comment:
maybe checksum.h/checksum.c is better to split semantically hashing (HTAB) and file checksums ? The same is for variables: S3FileHash -> S3FileChecksum.

I'm not sure about it, because it's already called hash in the previous code

Reply:
Indeed. I agree that "hash" might sound a bit ambiguous. I'll try to use "checksum".